src/importexport/musicxml/internal/musicxml/import/importmusicxmlpass2.cpp

```diff
@@ -8227,7 +8228,9 @@ static void addSlur(const Notation& notation, SlurStack& slurs, ChordRest* cr, c
             score->addElement(newSlur);
         } else {
             // slur start for new slur: init
-            Slur* newSlur = Factory::createSlur(score->dummy());
+            Slur* newSlur = notation.name()
+                            == "slur" ? Factory::createSlur(score->dummy())
+                            : static_cast<Slur*>(Factory::createHammerOnPullOff(score->dummy()));
```

Comment:
Un-optimal line break. And I guess the `static_cast` is unavoidable, right?

Also, perhaps it's safer to invert the condition, i.e. check for `"hammer-on"` or `"pull-off"` and by default just create a slur. (I don't like the fact that the function makes such an implicit assumption about where it is called and which input it can expect. But maybe that's too theoretical and maybe the MXML module is full of code that makes such assumptions, I don't have a good overview of it.)